### PR TITLE
do not pass mariadbd arguments with whitespace to mariadb-install-db

### PR DESCRIPTION
**Pull request by Alex1s**
Workaround for issue #574

### CODE REVIEW

File: 10.11/docker-entrypoint.sh [hunk 1]
```diff
@@ -227,8 +227,18 @@ _mariadb_version() {
...
-	# "Other options are passed to mariadbd." (so we pass all "mariadbd" arguments directly here)
-	mariadb-install-db "${installArgs[@]}" "${@:2}" \
+
+	mariadbdArgs=()
+	for arg in "${@:2}"; do
+		# Check if the argument contains whitespace
+		if [[ ! "$arg" =~ [[:space:]] ]]; then
+			mariadbdArgs+=("$arg")
+		else
+			mysql_warn Not passing argument \'$arg\' to mariadb-install-db because mariadb-install-db does not support arguments with whitespace.
```
Comment (HugoWenTD) [inline 10.11/docker-entrypoint.sh:237]: Should this be warning or error? Warnings could be silently ignored in the automation and lead to potential failures.

Comment (Alex1s) [reply 10.11/docker-entrypoint.sh:237]: Not sure if you are asking me ...
... but I would agree with you.

Comment (grooverdan) [reply 10.11/docker-entrypoint.sh:237]: I consider an error as something that cannot proceed and a warning is something that might be impacting.  As the original example showed there are some things that make little difference to the way mariadb-install-db processes.

Regardless I'm hoping that this is only a temporary change until the server implementation is made cleaner with some actual shell programming expertise.

Comment (fauust) [reply 10.11/docker-entrypoint.sh:237]: LGTM but I would rewrite the if:

```bash
if [[ "$arg" =~ [[:space:]] ]]; then
  warning
else
  OK
fi
```
seems much more simple to understand.

File: 10.11/docker-entrypoint.sh [hunk 2]
```diff
@@ -228,7 +228,17 @@ docker_init_database_dir() {
...
 	installArgs=( --datadir="$DATADIR" --rpm --auth-root-authentication-method=normal )
 	# "Other options are passed to mariadbd." (so we pass all "mariadbd" arguments directly here)
-	mariadb-install-db "${installArgs[@]}" "${@:2}" \
+
+	local mariadbdArgs=()
+	for arg in "${@:2}"; do
+		# Check if the argument contains whitespace
+		if [[ ! "$arg" =~ [[:space:]] ]]; then
+			mariadbdArgs+=("$arg")
+		else
```
Comment (Jesjimenez) [inline 10.11/docker-entrypoint.sh:237]: Cómo generó dinero

**Pull request merged**